optuna/samplers/_gp/_sampler.py

```diff
@@ -0,0 +1,169 @@
+    """Sampler using Gaussian process-based Bayesian optimization.
+
+    This sampler fits a Gaussian process (GP) to the objective function and optimizes
+    the acquisition function to suggest the next parameters.
+
+    The current implementation uses:
+    - Matern kernel with nu=2.5 (differentiable twice)
+    - Automatic relevance determination (ARD) for a lengthscale of each parameter
+    - Gamma prior for lengthscales, kernel scale, and noise scale
+    - Log Expected Improvement (logEI) as the acquisition function
+    - Quasi-Monte Carlo (QMC) sampling to optimize the acquisition function
+
+    .. note::
+        This sampler requires `scipy` and `pytorch` to be installed.
+        You can install these dependencies with `pip install scipy pytorch`.
+
+    Args:
+        seed:
+            Random seed passed to `independent_sampler`.
+            Defaults to None (random seed is used).
+
+        independent_sampler:
+            Sampler used for initial sampling (for the first `n_startup_trials` trials)
+            and for conditional parameters. Defaults to `None` (a random sampler is used).
+
+        n_startup_trials:
+            Number of initial trials. Defaults to 10.
+    """
+
+    def __init__(
+        self,
+        *,
+        seed: int | None = None,
+        independent_sampler: BaseSampler | None = None,
+        n_startup_trials: int = 10,
+    ) -> None:
+        self._independent_sampler = independent_sampler or optuna.samplers.RandomSampler(seed=seed)
+        self._intersection_search_space = optuna.search_space.IntersectionSearchSpace()
+        self._n_startup_trials = n_startup_trials
+        self._log_prior: "Callable[[_gp.KernelParams], torch.Tensor]" = default_log_prior
+        self._minimum_noise: float = 1e-6
+        self._last_kernel_params: "_gp.KernelParams | None" = None
+        self._optimize_n_samples: int = 2048
+
+    def reseed_rng(self) -> None:
+        self._independent_sampler.reseed_rng()
+
+    def infer_relative_search_space(
+        self, study: Study, trial: FrozenTrial
+    ) -> dict[str, BaseDistribution]:
+        return self._intersection_search_space.calculate(study)
+
+    def sample_relative(
+        self, study: Study, trial: FrozenTrial, search_space: dict[str, BaseDistribution]
+    ) -> dict[str, Any]:
+        if search_space == {}:
+            return {}
+
+        states = (TrialState.COMPLETE,)
+        trials = study._get_trials(deepcopy=False, states=states, use_cache=True)
+        if len(trials) < self._n_startup_trials:
+            return {}
+
+        (
+            internal_search_space,
+            transformed_params,
+        ) = _search_space.get_search_space_and_transformed_params(trials, search_space)
+        values = np.array([trial.value for trial in trials])
+        if study.direction == StudyDirection.MINIMIZE:
+            values = -values
+        values -= values.mean()
+
+        EPS = 1e-10
+        values /= max(EPS, values.std())
```

Comment:
[nit] I would like to name it `loss_vals` instead of `values` because `values` typically serves as a list of different objectives and we transform the values so that the value will be better when it is lower.

```suggestion
        minimize = study.direction == StudyDirection.MINIMIZE
        loss_vals = np.array([trial.value if minimize else -trial.value for trial in trials])
        loss_vals = (loss_vals - np.mean(loss_vals)) / max(1e-10, np.std(values))
```

Reply:
We actually transform it so that higher is better (which is the default notation of GP, cf. acquisition function)